sybilCredentialAtomicMTP.go

```diff
@@ -0,0 +1,262 @@
+	"encoding/json"
+	"fmt"
+	core "github.com/iden3/go-iden3-core"
+	"github.com/iden3/go-merkletree-sql/v2"
+	"github.com/pkg/errors"
+	"math/big"
+	"strconv"
+)
+
+type SybilAtomicMTPInputs struct {
+	BaseConfig
+
+	ID                       *core.ID
+	ProfileNonce             *big.Int
+	ClaimSubjectProfileNonce *big.Int
+
+	IssuerClaim          ClaimWithMTPProof
+	StateCommitmentClaim ClaimWithMTPProof
+
+	GISTProof GISTProof
+	CRS       *big.Int
+
+	RequestID *big.Int
+	Timestamp int64
+}
+
+type sybilAtomicMTPCircuitInputs struct {
+	IssuerClaim           *core.Claim        `json:"issuerClaim"`
+	IssuerClaimMtp        []*merkletree.Hash `json:"issuerClaimMtp"`
+	IssuerClaimClaimsRoot *merkletree.Hash   `json:"issuerClaimClaimsRoot"`
+	IssuerClaimRevRoot    *merkletree.Hash   `json:"issuerClaimRevRoot"`
+	IssuerClaimRootsRoot  *merkletree.Hash   `json:"issuerClaimRootsRoot"`
+	IssuerClaimIdenState  *merkletree.Hash   `json:"issuerClaimIdenState"`
+
+	IssuerClaimNonRevMtp      []*merkletree.Hash `json:"issuerClaimNonRevMtp"`
+	IssuerClaimNonRevMtpNoAux string             `json:"issuerClaimNonRevMtpNoAux"`
+	IssuerClaimNonRevMtpAuxHi *merkletree.Hash   `json:"issuerClaimNonRevMtpAuxHi"`
+	IssuerClaimNonRevMtpAuxHv *merkletree.Hash   `json:"issuerClaimNonRevMtpAuxHv"`
+
+	IssuerClaimNonRevClaimsRoot *merkletree.Hash `json:"issuerClaimNonRevClaimsRoot"`
+	IssuerClaimNonRevRevRoot    *merkletree.Hash `json:"issuerClaimNonRevRevRoot"`
+	IssuerClaimNonRevRootsRoot  *merkletree.Hash `json:"issuerClaimNonRevRootsRoot"`
+	IssuerClaimNonRevState      *merkletree.Hash `json:"issuerClaimNonRevState"`
+
+	ClaimSchema string `json:"claimSchema"`
+
+	StateCommitmentClaim           *core.Claim        `json:"stateCommitmentClaim"`
+	StateCommitmentClaimMtp        []*merkletree.Hash `json:"stateCommitmentClaimMtp"`
+	StateCommitmentClaimClaimsRoot *merkletree.Hash   `json:"stateCommitmentClaimClaimsRoot"`
+	StateCommitmentClaimRevRoot    *merkletree.Hash   `json:"stateCommitmentClaimRevRoot"`
+	StateCommitmentClaimRootsRoot  *merkletree.Hash   `json:"stateCommitmentClaimRootsRoot"`
+	StateCommitmentClaimIdenState  *merkletree.Hash   `json:"stateCommitmentClaimIdenState"`
+
+	GistRoot     *merkletree.Hash   `json:"gistRoot"`
+	GistMtp      []*merkletree.Hash `json:"gistMtp"`
+	GistMtpAuxHi *merkletree.Hash   `json:"gistMtpAuxHi"`
+	GistMtpAuxHv *merkletree.Hash   `json:"gistMtpAuxHv"`
+	GistMtpNoAux string             `json:"gistMtpNoAux"`
+
+	CRS *big.Int `json:"crs"`
+
+	UserGenesisID            string `json:"userGenesisID"`
+	ProfileNonce             string `json:"profileNonce"`
+	ClaimSubjectProfileNonce string `json:"claimSubjectProfileNonce"`
+
+	RequestID *big.Int `json:"requestID"`
+	IssuerID  string   `json:"issuerID"`
+	Timestamp int64    `json:"timestamp"`
+}
+
+func (s SybilAtomicMTPInputs) Validate() error {
+
+	if s.ID == nil {
+		return errors.New(ErrorEmptyID)
+	}
+
+	if s.GISTProof.Proof == nil {
+		return errors.New(ErrorEmptyGlobalProof)
```

Comment:
Replace `ErrorEmptyGlobalProof` with `ErrorEmptyGISTProof` maybe?

Reply:
I didnt create the error just used it, you think I should update its name?

Reply:
updated according to your other comment (all names with global have been changed to GIST